supabase/functions/oauth/encrypt-config.ts

```diff
@@ -48,19 +50,23 @@ export async function encryptConfig(req: Record<string, any>) {
   });
 
   let responseData = JSON.stringify(await response.json());
-  console.log(responseData);
 
   // If we can find client_id or client_secret in plaintext in the response,
   // it's not secure to return this response!
   if (
-    responseData.includes(oauth2_client_id) ||
-    responseData.includes(oauth2_client_secret)
+    oauth2_client_id != null &&
```

Comment:
This code path is actually really important, because the JSON schema is taken from the request. Since an attacker could control the schema, they could for example remove the `secret` annotations and exfiltrate our client id and secret.

I actually think the previous conditional is preferable here, as well. If we could ever end up in a situation where only one of client_id or client_secret are non null, then we would skip this important check. It's _probably_ the case that client_id and secret will always come together or not at all, but it costs us essentially nothing to just always check if the id/secret are there and error out.